lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -234,6 +244,39 @@ monitorStakePools tr gp nl db@DBLayer{..} = do
                 putPoolRegistration slot pool
         pure Continue
 
+monitorMetadata
+    :: Tracer IO StakePoolLog
+    -> GenesisParameters
+    -> (StakePoolMetadataUrl -> StakePoolMetadataHash -> IO (Either String StakePoolMetadata))
+    -> DBLayer IO
+    -> IO ()
+monitorMetadata tr gp fetchMetadata DBLayer{..} = forever $ do
+    refs <- atomically (unfetchedPoolMetadataRefs 100)
+
+    successes <- fmap catMaybes $ forM refs $ \(pid, url, hash) -> do
+        traceWith tr $ MsgFetchPoolMetadata pid url
+        fetchMetadata url hash >>= \case
+            Left msg -> Nothing <$ do
+                traceWith tr $ MsgFetchPoolMetadataFailure pid msg
+
+            Right meta -> Just pid <$ do
+                traceWith tr $ MsgFetchPoolMetadataSuccess pid meta
+                atomically $ putPoolMetadata pid hash meta
+
+    when (null refs || null successes) $ do
+        traceWith tr $ MsgFetchTakeBreak blockFrequency
+        threadDelay blockFrequency
+  where
+    -- NOTE
+    -- If there's no metadata, we typically need not to retry sooner than the
+    -- next block. So waiting for a delay that is roughly the same order of
+    -- magnitude as the (slot length / active slot coeff) sounds sound.
+    blockFrequency = ceiling (1/f) * toMicroSecond slotLength
```

Comment:
Don't you want a `$` after `ceiling`?
```haskell
λ> f
9.0e-2
λ> sl
10
λ> (1/f) * sl
111.11111111111111
λ> ceiling (1/f) * sl
120
```

Reply:
This is an order of magnitude, so it really doesn't matter.